modules/navigation/3d/nav_mesh_queries_3d.cpp

```diff
@@ -157,6 +157,18 @@ void NavMeshQueries3D::map_query_path(NavMap *map, const Ref<NavigationPathQuery
 	query_task.target_position = p_query_parameters->get_target_position();
 	query_task.navigation_layers = p_query_parameters->get_navigation_layers();
 	query_task.callback = p_callback;
+	query_task.exclude_regions = p_query_parameters->get_excluded_regions().size() > 0;
+	if (query_task.exclude_regions) {
+		for (const RID region_rid : p_query_parameters->get_excluded_regions()) {
```

Comment:
Not very important as this will probably never be really big, but you could resize `query_task.excluded_regions` first to the final size, to be sure to only have one allocation, instead of maybe 2 or 3.